en/ros/ros2_comm.md

```diff
@@ -0,0 +1,579 @@
+# PX4-ROS 2 Bridge
+
+This section of the guide focusses on the bridge between PX4 and ROS 2, made available through the *PX4-Fast RTPS(DDS) Bridge* and resulting products to interface with ROS 2, *px4_ros_com* and *px4_msgs* ROS 2 packages.
+
+It also provides information in how to:
+1. Connect ROS 2 nodes with PX4 (via the *microRTPS* Bridge, and using the `px4_ros_com` package)
+1. Connect ROS (ROS "version 1") nodes with PX4 by additionally using the `ros1_bridge` package to bridge ROS 2 and ROS (1).
+
+:::note
+For information about using the *PX4-Fast RTPS(DDS) Bridge* without ROS 2, see the [RTPS/DDS Interface section](../middleware/micrortps.md).
+:::
+
+:::note
+For a more detailed and visual explanation of using the bridge with ROS 2, see the following talks/presentation videos:
+1. [PX4 Dev Summit 2019 - "ROS 2 Powered PX4"](https://www.youtube.com/watch?v=2Szw8Pk3Z0Q)
+1. [ROS World 2020 - Getting started with ROS 2 and PX4](https://www.youtube.com/watch?v=qhLATrkA_Gw)
+:::
+
+## Why Two Bridges (PX4/ROS2 vs PX4/FastDDS)?
+
+The [PX4-Fast RTPS(DDS) Bridge](../middleware/micrortps.md) and PX4-ROS2 bridge (described here) are conceptually the same.
+- They both have a client running on PX4 that communicates with an agent running on a companion computer.
+- The PX4 firmware build process creates the client code.
+- The PX4 firmware build process creates IDL files matching a select set of UORB topics that will be shared over DDS.
+- These IDL files are then used by code generators to create the Agent source code, which can then be compiled for the companion. 
+
+The difference is that ROS 2 uses slightly different DDS types than "raw" RTPS/DDS applications.
+These types require slighty different agent code, generated from slightly different IDL files.
+The client-side code is the same in both cases.
+
+The PX4 build process automatically generates the IDL file-format required for ROS 2 and publishes them to `px4_msgs`.
```

Comment:
```suggestion
The PX4 build process creates ROS2 message files for each uORB message and deploys them to the `px4_msgs` repo. Based on those `.msg` files, the `px4_msgs` package build will generate the ROS 2 interface and typesupport code to be used by ROS2 nodes, but also simultaneously generates the IDL files that are going to be consumed by the code generators in `px4_ros_com` to generate the agent source code compatible with ROS 2.
```

Reply:
I merged a slight variation of this. Thanks!